Node perf tester

TimvdLippe opened the pull request:
Add a node interface for the perf-tester. It uses chrome headless if no port is specified. It parses a series of options which it will navigate to, perform the timings and then extract `window.perfTiming` from the `Runtime`.

TimvdLippe committed:
Add node perf tester based on Chrome Devtools protocol

TimvdLippe committed:
Fix missing window.perfTiming and correct line-endings

sorvell commented on options.json:
```diff
@@ -0,0 +1,4 @@
+[
+    ["original", "http://localhost:8081/youtube/"],
```
These should be removed, right?

sorvell commented on perf.js:
```diff
@@ -1,57 +1,59 @@
-    }
-  };
-
+// x-browser compat.
```
The diff here is weird. What happened?

TimvdLippe replied:
The line endings were mixed. I changed all lines to LF.

TimvdLippe committed:
Address reviewer feedback

TimvdLippe committed:
Fix formatting

TimvdLippe commented:
@sorvell I updated the line endings, so the diff is less nice :cry: I updated `console._perfEnd` to take an options object to disable the force layout, as well as expose the actual timing in `window.perfTiming`, such that the node script can use it. I also added some documentation on how to use the script.

hastebrot commented:
>I updated the line endings, so the diff is less nice

Using `w=1` (ignore whitespace; similar to `git diff --ignore-all-space`) in the URL for the diff viewer (compare) helps a bit:
https://github.com/PolymerLabs/perf-tester/pull/4/files?diff=unified&w=1
